core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -3196,12 +3197,40 @@ public RelBuilder sort(RelCollation collation) {
    */
   public RelBuilder sortLimit(int offset, int fetch,
       Iterable<? extends RexNode> nodes) {
+    final @Nullable RexNode offsetNode = offset <= 0 ? null : literal(offset);
+    final @Nullable RexNode fetchNode = fetch < 0 ? null : literal(fetch);
+    if (offsetNode == null && fetch == 0 && config.simplifyLimit()) {
```

Comment:
Sorry, I know I was the one that proposed this code, but I think we should not include this `if (...) return empty()` simplification, because it will be a duplication (this same simplification will be part any way of the overloaded method with RexNode parameters, so I think we should remove it from this one).

Reply:
Never mind, I deleted the duplicate code. Thank you again~